src/cpp/session/projects/SessionProjects.cpp

```diff
@@ -325,24 +325,50 @@ Error createProjectFile(const json::JsonRpcRequest& request,
 {
    using namespace projects;
    
-   std::string projFile;
-   Error error = json::readParams(request.params, &projFile);
+   std::string projDir;
+   Error error = json::readParams(request.params, &projDir);
    if (error)
       return error;
    
-   FilePath projPath = module_context::resolveAliasedPath(projFile);
-   if (!projPath.exists())
+   // Resolve the path and ensure it exists
+   FilePath projDirPath = module_context::resolveAliasedPath(projDir);
+   FilePath projFilePath;
+   if (!projDirPath.exists())
    {
-      Error error = r_util::writeProjectFile(
-               projPath,
+      return pathNotFoundError(projDir, ERROR_LOCATION);
+   }
+
+   // Check for an existing project file in the directory
```

Comment:
What about using `projectFromDirectory()` here? https://github.com/rstudio/rstudio/blob/02ee571510791d983989f5c70233293d72123336/src/cpp/core/r_util/RProjectFile.cpp#L1014-L1080

Reply:
In particular it looks like we handle the case where there are multiple `.Rproj` files but one is newer than the others.

Reply:
Good idea!